gcloud-java-compute/src/test/java/com/google/gcloud/compute/ComputeImplTest.java

```diff
@@ -1425,6 +1457,445 @@ public void testDeleteZoneOperation_False() {
+    assertNull(page.nextPageCursor());
+    assertArrayEquals(addresses.toArray(), Iterables.toArray(page.values(), Address.class));
+  }
+
+  @Test
+  public void testListGlobalAddressesWithOptions() {
+    String cursor = "cursor";
+    compute = options.service();
+    ImmutableList<Address> addressList = ImmutableList.of(
+        new Address(compute, new AddressInfo.BuilderImpl(GLOBAL_ADDRESS)),
+        new Address(compute, new AddressInfo.BuilderImpl(GLOBAL_ADDRESS)));
+    Tuple<String, Iterable<com.google.api.services.compute.model.Address>> result =
+        Tuple.of(cursor, Iterables.transform(addressList, AddressInfo.TO_PB_FUNCTION));
+    EasyMock.expect(computeRpcMock.listGlobalAddresses(ADDRESS_LIST_OPTIONS)).andReturn(result);
+    EasyMock.replay(computeRpcMock);
+    Page<Address> page = compute.listGlobalAddresses(ADDRESS_LIST_PAGE_SIZE,
+        ADDRESS_LIST_PAGE_TOKEN, ADDRESS_LIST_FILTER);
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(addressList.toArray(), Iterables.toArray(page.values(), Address.class));
+  }
+
+  @Test
+  public void testListRegionAddresses() {
+    String cursor = "cursor";
+    compute = options.service();
+    ImmutableList<Address> addressList = ImmutableList.of(
+        new Address(compute, new AddressInfo.BuilderImpl(REGION_ADDRESS)),
+        new Address(compute, new AddressInfo.BuilderImpl(REGION_ADDRESS)));
+    Tuple<String, Iterable<com.google.api.services.compute.model.Address>> result =
+        Tuple.of(cursor, Iterables.transform(addressList,
+            new Function<Address, com.google.api.services.compute.model.Address>() {
+              @Override
+              public com.google.api.services.compute.model.Address apply(Address address) {
+                return address.toPb();
+              }
+            }));
+    EasyMock.expect(
+        computeRpcMock.listRegionAddresses(REGION_ADDRESS_ID.region(), EMPTY_RPC_OPTIONS))
+        .andReturn(result);
+    EasyMock.replay(computeRpcMock);
+    Page<Address> page = compute.listRegionAddresses(REGION_ADDRESS_ID.region());
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(addressList.toArray(), Iterables.toArray(page.values(), Address.class));
+  }
+
+  @Test
+  public void testListRegionAddressesNextPage() {
+    String cursor = "cursor";
+    String nextCursor = "nextCursor";
+    compute = options.service();
+    ImmutableList<Address> addressList = ImmutableList.of(
+        new Address(compute, new AddressInfo.BuilderImpl(REGION_ADDRESS)),
+        new Address(compute, new AddressInfo.BuilderImpl(REGION_ADDRESS)));
+    ImmutableList<Address> nextAddressList = ImmutableList.of(
+        new Address(compute, new AddressInfo.BuilderImpl(REGION_ADDRESS)));
+    Tuple<String, Iterable<com.google.api.services.compute.model.Address>> result =
+        Tuple.of(cursor, Iterables.transform(addressList,
+            new Function<Address, com.google.api.services.compute.model.Address>() {
```

Comment:
It seems like this function is used multiple times.  Do you think it's worth making it global?